pkg/collector/corechecks/cluster/kubernetes_apiserver.go

```diff
@@ -0,0 +1,158 @@
+	return yaml.Unmarshal(data, c)
+}
+
+// Configure parses the check configuration and init the check.
+func (k *KubeASCheck) Configure(config, initConfig check.ConfigData) error {
+	// Check connectivity to the APIServer
+	err := k.instance.parse(config)
+	if err != nil {
+		log.Error("could not parse the config for the API server")
+		return err
+	}
+
+	log.Debugf("Running config %s", config)
+	return nil
+}
+
+// GetWarnings grabs the last warnings from the sender.
+func (k *KubeASCheck) GetWarnings() []error {
+	w := k.lastWarnings
+	k.lastWarnings = []error{}
+	return w
+}
+
+// ID returns the name of the check since there should be only one instance running.
+func (k *KubeASCheck) ID() check.ID {
+	return check.ID(k.String())
+}
+
+// Interval returns the scheduling time for the check.
+func (k *KubeASCheck) Interval() time.Duration {
+	return check.DefaultCheckInterval
+}
+
+// Run executes the check.
+func (k *KubeASCheck) Run() error {
+	sender, err := aggregator.GetSender(k.ID())
+	if err != nil {
+		return err
+	}
+
+	asclient, err := as_util.GetAPIClient()
+
+	if err != nil {
+		log.Errorf("could not instantiate the cluster %q", err)
+		return err
+	}
+
+	componentsStatus := asclient.GetComponents()
+
+	k.parseComponentStatus(sender, componentsStatus)
+
+	return nil
+}
+
+// Stop does nothing.
+func (k *KubeASCheck) Stop() {}
+
+// KubernetesASFactory is exported for integration testing.
+func KubernetesASFactory() check.Check {
+	return &KubeASCheck{
+		instance: &KubeASConfig{},
+	}
+}
+
+// GetMetricStats returns the stats from the last run of the check.
+func (k *KubeASCheck) GetMetricStats() (map[string]int64, error) {
+	sender, err := aggregator.GetSender(k.ID())
+	if err != nil {
+		return nil, log.Errorf("Failed to retrieve a Sender instance: %v", err)
+	}
+	return sender.GetMetricStats(), nil
+}
+
+func (k *KubeASCheck) parseComponentStatus(sender aggregator.Sender, componentsStatus *v1.ComponentStatusList) {
+
+	for _, component := range componentsStatus.Items {
+
+		if component.Metadata.Name == nil || component.Conditions == nil {
+			log.Error("Metadata structure has changed. Not collecting API Server's Components status")
+			return
+		}
+		tagComp := append(k.instance.Tags, fmt.Sprintf("component:%s", *component.Metadata.Name))
+		// We only expect the Healthy condition. May change in the future. https://godoc.org/github.com/ericchiang/k8s/api/v1#ComponentCondition
+		for _, condition := range component.Conditions {
+			// We only expect True, False and Unknown.
+			switch {
+			case *condition.Type != "Healthy":
+				log.Debugf("Condition %q not supported", *condition.Type)
+				sender.ServiceCheck(KubeControlPaneCheck, metrics.ServiceCheckUnknown, k.KubeAPIServerHostname, tagComp, "The Component's condition type isn't supported")
+				continue
+
+			case *condition.Status == "True":
+				sender.ServiceCheck(KubeControlPaneCheck, metrics.ServiceCheckOK, k.KubeAPIServerHostname, tagComp, "")
```

Comment:
- It would be great for maintainability to only have one `sender.ServiceCheck` invocation, and prepare the SC condition before
- We could get the `.Message` field if not nil and use it as SC messsage
- You could comment in the `default` case we catch the "unknown" value plus possibly new ones